modules/cuckooml/cuckooml.py

```diff
@@ -0,0 +1,1677 @@
+        pd.DataFrame(clf).to_csv("clustering_fit.csv", encoding="utf-8")
+
+        # Save clustering results to csv
+        print "Clustering results saved to `clustering_results.csv`"
+        ml.clustering[cls_alg]["clustering"].to_csv("clustering_results.csv", \
+                                                    encoding="utf-8")
+        print "Cluster label distribution saved to \
+               `cluster_label_distribution.csv`"
+        ml.clustering_label_distribution(ml.clustering[cls_alg]["clustering"], \
+                                         ml.labels, plot=False)\
+            .to_csv("cluster_label_distribution.csv", encoding="utf-8")
+
+    # Save abnormal behaviour stats?
+    if cfg.cuckooml.abnormal_behaviour:
+        print "Abnormal behaviour will be saved to `abnormal_behaviour.csv`"
+        ml.detect_abnormal_behaviour(features_numerical, figures=False).to_csv(\
+            "abnormal_behaviour.csv", encoding="utf-8")
+
+    if cfg.cuckooml.anomalies_detection:
+        ad = pd.DataFrame(ml.clustering[cls_alg]["clustering"]["label"])
+        ad.columns = ["cluster"]
+        ad_dict = ml.anomaly_detection()
+        for i in ad_dict:
+            if isinstance(ad_dict[i], list):
+                ad[i] = pd.Series([1]*len(ad_dict[i]), index=ad_dict[i])
+            elif isinstance(ad_dict[i], dict):
+                in_list = []
+                for j in ad_dict[i]:
+                    in_list += ad_dict[i][j]
+                ad[i] = pd.Series([1]*len(in_list), index=in_list)
+        print "Saving anomaly detection results as `anomalies.csv`"
+        ad.to_csv("anomalies.csv", encoding="utf-8")
+
+
+    if cfg.cuckooml.compare_new_samples:
+        test_location = CUCKOO_ROOT + "/" + cfg.cuckooml.test_directory
+
+        if os.path.isdir(test_location):
+            new_sample = Loader()
+            new_sample.load_binaries(test_location)
+        elif os.path.isfile(test_location):
+            new_sample = Instance()
+            new_sample.load_json(test_location)
+            new_sample.label_sample()
+            new_sample.extract_features()
+            new_sample.extract_basic_features()
+        # Compare new sample(s)
+        ml.compare_sample(new_sample).to_csv("test_samples.csv", \
+                                             encoding="utf-8")
+
+    if cfg.cuckooml.clustering and cfg.cuckooml.save_clustering_results:
+        if cfg.cuckooml.clustering_results_directory:
+            ml.save_clustering_results(loader, CUCKOO_ROOT+"/"+cfg.cuckooml.\
+                                       clustering_results_directory)
+        else:
+            ml.save_clustering_results(loader)
+
+
+class ML(object):
+    """Feature formatting and machine learning for Cuckoo analysed binaries.
+    All functions marked with asterisk (*) were inspired by code distributed
+    with "Back to the Future: Malware Detection with Temporally Consistent
+    Labels" by Brad Miller at al."""
+    SIMPLE_CATEGORIES = {
```

Comment:
@jbremer As far as I can understand you want to detach the *inspired* code from cuckooml and put it in the `cuckoosandbox/community` repository; is that correct?
If so it would probably only make it more difficult to use, wouldn't it?